### PR TITLE
Eth and personal sign

### PR DESCRIPTION
**Pull request by estebanmino**
This PR is mostly a migration to ts of [this](https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/lib/personal-message-manager.js) to have both eth_sign and personal_sign as proposed here https://github.com/MetaMask/metamask-extension/pull/4117

It also exposes `signPersonalMessage` from `eth-keyring-controller`

### CODE REVIEW

File: src/PersonalMessageManager.ts [hunk 1]
```diff
@@ -0,0 +1,332 @@
+import { EventEmitter } from 'events';
+import BaseController, { BaseConfig, BaseState } from './BaseController';
+import { validatePersonalSignMessageData, normalizeMessageData } from './util';
+const random = require('uuid/v1');
```
Comment (brunobar79) [inline src/PersonalMessageManager.ts]: import ?

File: src/PersonalMessageManager.ts [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+
+/**
+ * @type Message
+ *
+ * Represents, and contains data about, an 'personal_sign' type signature request.
```
Comment (brunobar79) [inline src/PersonalMessageManager.ts]: "Represents, and contains data about a 'personal_sign' type signature request."

File: src/PersonalMessageManager.ts [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+ * These are created when a signature for an personal_sign call is requested.
+ *
+ * @property id - An id to track and identify the message object
+ * @property messageParams - The parameters to pass to the personal_sign method once the signature request is approved
+ * @property type - The json-prc signing method for which a signature request has been made.
+ * A 'Message' with always have a 'personal_sign' type
```
Comment (brunobar79) [inline src/PersonalMessageManager.ts]: "A 'Message' which always has..."

File: src/PersonalMessageManager.ts [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+export interface MessageParams {
+	data: string;
+	from: string;
+	origin?: string;
+}
+
+/**
+ * @type MessageParamsMetamask
+ *
+ * Represents, the parameters to pass to the personal_sign method once the signature request is approved
```
Comment (brunobar79) [inline src/PersonalMessageManager.ts]: Represents  the (remove comma)

File: src/PersonalMessageManager.ts [hunk 5]
```diff
@@ -0,0 +1,332 @@
...
+	}
+
+	/**
+	 * A getter for the 'unapproved' Messages in state messages
+	 *
+	 * @returns - An index of Message ids to Messages, for all 'unapproved' Messages in this.messages
+	 *
+	 */
+	getUnapprovedMessages() {
+		return this.messages
```
Comment (brunobar79) [inline src/PersonalMessageManager.ts]: ES6 + Functional programming + TypeScript = 💣 🔥

File: src/PersonalMessageManager.ts (hunk 1 above)
Comment (estebanmino) [reply src/PersonalMessageManager.ts]: honestly I didn't question this, TransactionController uses the same and imports it in the same way

**Pull request merged**